core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -810,12 +826,17 @@ boolean tooHigh(Object key) {
                     (c == 0 && !hiInclusive)));
         }
 
+
         boolean inBounds(Object key) {
-            return !tooLow(key) && !tooHigh(key);
+            if (!isDescending) {
```

Comment:
you have assumption on where it is called
need to document this assumption
and make this method private